nmigen/vendor/xilinx_7series.py

```diff
@@ -165,17 +312,27 @@ def create_missing_domain(self, name):
                 rst_i = self.request(self.default_rst).i
 
             m = Module()
-            ready = Signal()
-            m.submodules += Instance("STARTUPE2", o_EOS=ready)
-            m.domains += ClockDomain("sync", reset_less=self.default_rst is None)
-            # Actually use BUFGCTRL configured as BUFGCE, since using BUFGCE causes sim/synth
-            # mismatches with Vivado 2019.2, and the suggested workaround (SIM_DEVICE parameter)
-            # breaks Vivado 2017.4.
-            m.submodules += Instance("BUFGCTRL",
-                i_I0=clk_i,   i_S0=C(1, 1), i_CE0=ready,   i_IGNORE0=C(0, 1),
-                i_I1=C(1, 1), i_S1=C(0, 1), i_CE1=C(0, 1), i_IGNORE1=C(1, 1),
-                o_O=ClockSignal("sync")
-            )
+
+            if self.toolchain == "Vivado":
+                ready = Signal()
+                m.submodules += Instance("STARTUPE2", o_EOS=ready)
+                m.domains += ClockDomain("sync", reset_less=self.default_rst is None)
+                # Actually use BUFGCTRL configured as BUFGCE, since using BUFGCE causes sim/synth
+                # mismatches with Vivado 2019.2, and the suggested workaround (SIM_DEVICE parameter)
+                # breaks Vivado 2017.4.
+                m.submodules += Instance("BUFGCTRL",
+                    i_I0=clk_i,   i_S0=C(1, 1), i_CE0=ready,   i_IGNORE0=C(0, 1),
+                    i_I1=C(1, 1), i_S1=C(0, 1), i_CE1=C(0, 1), i_IGNORE1=C(1, 1),
+                    o_O=ClockSignal("sync")
+                )
+            elif self.toolchain == "Symbiflow":
+                cd_sync = ClockDomain("sync", reset_less=self.default_rst is None)
+                m.domains += cd_sync
+                m.submodules += Instance("BUFG", i_I=clk_i, o_O=cd_sync.clk)
```

Comment:
Does Symbiflow actually *need* this `BUFG`, or can it insert one automatically?

Reply:
Simple example (blinky) works without this bufg. However, we had some problems with Litex+VexriscV+Linux without it.